File: 2089-maximum-matrix-sum/maximum-matrix-sum.py
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        total = 0
        neg_count = 0
        mini = float("inf")
        for r in matrix:
            for c in r:
                total += abs(c)
                mini = min(mini, abs(c))
                if c < 0:
                    neg_count += 1
        return total if neg_count % 2 == 0 else total - (mini * 2)

File: 2089-maximum-matrix-sum/maximum-matrix-sum.java
class Solution {
    public long maxMatrixSum(int[][] matrix) {
        long sum = 0;
        int mini = Integer.MAX_VALUE;
        int neg_count = 0;
        for(int[] r: matrix)
            for(int c: r)
            {
                sum += Math.abs(c);
                mini = Math.min(mini, Math.abs(c));
                if (c < 0)
                    neg_count++;
            }
        return neg_count % 2 == 0 ? sum : sum - (mini * 2); 
    }
}